File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int sol[] = new int[nums.length];

        int start = 0;
        int end = nums.length - 1;
        int idx = sol.length - 1;

        while (start <= end) {
            if (nums[start] * nums[start] >= nums[end] * nums[end]) {
                sol[idx] = nums[start] * nums[start];
                start++;
            } else {
                sol[idx] = nums[end] * nums[end];
                end--;
            }
            idx--;
        }
        return sol;
    }
}

File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        l = len(nums)
        for i in range(0,l):
            nums[i] = nums[i] * nums[i]
            if nums[i] < 0:
                nums[i] = nums[i] + nums[i]
        nums.sort()
        return nums
        